src/Compilers/CSharp/Portable/Binder/Binder_Attributes.cs

```diff
@@ -501,10 +501,19 @@ protected bool IsAttributeConditionallyOmitted(NamedTypeSymbol attributeType, Sy
 
         private BoundAssignmentOperator BindNamedAttributeArgument(AttributeArgumentSyntax namedArgument, NamedTypeSymbol attributeType, BindingDiagnosticBag diagnostics)
         {
+            Debug.Assert(namedArgument.NameEquals is not null);
+            IdentifierNameSyntax nameSyntax = namedArgument.NameEquals.Name;
+
+            if (attributeType.IsErrorType())
+            {
+                var badLHS = BadExpression(nameSyntax, lookupResultKind: LookupResultKind.Empty);
+                var badRHS = this.BindValue(namedArgument.Expression, diagnostics, BindValueKind.RValue);
```

Comment:
This needs a `BindToNaturalTypeForErrorRecovery`. Please add a test with a target-typed node such as a switch expression, and pull on the value through some path that will fail if it's not bound to a real type, such as IOperation.

Reply:
Good catch. I thought the `null` already covered the scenario for typeless expressions, but the switch indeed asserted. Thanks